### PR TITLE
Bundle Charge Links Commands

### PR DESCRIPTION
**Pull request by Mech0z**
**Description**
We want to support bundled ingestion messages in the charges domain so that market participants are able to send multiple charge links in one message to the charge domain.

**Tech note**
It's currently not possible to create a reject flow until #938 is finished, this story will expand the domain tests with rejected charge links

**Acceptance Criteria:**
Given a Grid Access Provider submits a bundled charge links message containing valid charge links requests
When processed by the Charges domain
Then each charge link (in the request) is stored
And a confirmation is made available for each charge link (in the request) to the Grid Access Provider

**Out of scope**
It is ensured that changes in a link bundle which affect the same combination of metering point and charge, are applied in the same order as they appeared in the bundled message

## References

* #919

### CODE REVIEW

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/DomainTests/ChargeLinkIngestionTests.cs [hunk 1]
```diff
@@ -67,11 +67,12 @@ public async Task Given_NewTaxChargeLinkMessage_When_GridAccessProviderPeeks_The
                 var request = CreateHttpRequest(ChargeLinkDocument.AnyTax, out var correlationId);
 
                 // Act
-                await Fixture.HostManager.HttpClient.SendAsync(request);
+                var response = await Fixture.HostManager.HttpClient.SendAsync(request);
 
                 // Assert
                 // We expect two message types in the messagehub, one for the receipt and one for the charge link itself
                 await Fixture.MessageHubMock.AssertPeekReceivesReplyAsync(correlationId, 2);
+                response.IsSuccessStatusCode.Should().BeTrue();
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/DomainTests/ChargeLinkIngestionTests.cs:75]: Is this assert necessary? The test above does check for a 200 OK response.

Comment (Mech0z) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/DomainTests/ChargeLinkIngestionTests.cs]: removed

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Application/ChargeLinks/Handlers/ChargeLinksCommandHandlerTests.cs [hunk 2]
```diff
@@ -29,14 +32,18 @@ public class ChargeLinksCommandHandlerTests
         [Theory]
         [InlineAutoDomainData]
         public async Task HandleAsync_WhenCalledWithValidChargeLink_ShouldReturnOk(
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Application/ChargeLinks/Handlers/ChargeLinksCommandHandlerTests.cs:34]: And this test name doesn't seem to match the test. How about:
```suggestion
        public async Task HandleAsync_WhenValidChargeLinkCommand_DispatchesOnce(
```

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Infrastructure/CimDeserialization/ChargeLinkBundle/ChargeLinkCommandConverterTests.cs [hunk 3]
```diff
@@ -114,16 +116,69 @@ public class ChargeLinkCommandConverterTests
...
 
             // Act
-            var result = (ChargeLinksCommand)await sut.ConvertAsync(reader).ConfigureAwait(false);
+            var resultBundle = (ChargeLinksCommandBundle)await sut.ConvertAsync(reader).ConfigureAwait(false);
 
             // Assert
-            Assert.Equal("DocId_Valid_003", result.Document.Id);
-            Assert.Equal("rId_Valid_003", result.ChargeLinks.First().OperationId);
-            Assert.Null(result.ChargeLinks.First().EndDateTime);
+            var result = resultBundle.ChargeLinksCommands.First();
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Infrastructure/CimDeserialization/ChargeLinkBundle/ChargeLinkCommandConverterTests.cs:122]: nit: As we know there is exactly one charge link command in the test file:
```suggestion
            var result = resultBundle.ChargeLinksCommands.Single();
```

**Review by prtandrup — APPROVED** [review]
Approved with a few comments

**Pull request merged**